<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 5 Lab: GeneMANIA (Cytoscape version) | Pathway and Network Analysis of -Omics Data ( November 2024 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 5 Lab: GeneMANIA (Cytoscape version) | Pathway and Network Analysis of -Omics Data ( November 2024 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 5 Lab: GeneMANIA (Cytoscape version) | Pathway and Network Analysis of -Omics Data ( November 2024 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Gary Bader, Ruth Isserlin, Chaitra Sarathy, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-5-gene-function-prediction.html"/>
<link rel="next" href="genemania_web.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Pathways Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Canadian Bioinformatics Workshops</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gary-bader"><i class="fa fa-check"></i>Gary Bader</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#lincoln-stein"><i class="fa fa-check"></i>Lincoln Stein</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gregory-schwartz"><i class="fa fa-check"></i>Gregory Schwartz</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#chaitra-sarathy-phd"><i class="fa fa-check"></i>Chaitra Sarathy, PhD</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#nia-hughes"><i class="fa fa-check"></i>Nia Hughes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#class-materials"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#docker-installation"><i class="fa fa-check"></i>Docker Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#r-tutorial"><i class="fa fa-check"></i>R Tutorial</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-9-optional-by-recommended"><i class="fa fa-check"></i>Step 9 (Optional by recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-1"><i class="fa fa-check"></i><strong>Optional 1</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-2"><i class="fa fa-check"></i><strong>Option 2</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-3"><i class="fa fa-check"></i><strong>Option 3</strong> :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html"><i class="fa fa-check"></i>Module 3 Lab Primer: Cytoscape Primer</a>
<ul>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1a---create-network-from-table"><i class="fa fa-check"></i>Exercise 1a - Create Network from table</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1b---load-node-attributes"><i class="fa fa-check"></i>Exercise 1b - Load node attributes</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1c---map-node-attributes-to-visual-style"><i class="fa fa-check"></i>Exercise 1c - Map node attributes to Visual Style</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-2---work-with-larger-networks"><i class="fa fa-check"></i>Exercise 2 - Work with larger networks</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-3---perform-basic-enrichment-analysis-using-enrichmenttable"><i class="fa fa-check"></i>Exercise 3 - Perform basic enrichment analysis using EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#enrichmenttabl-features"><i class="fa fa-check"></i>Exercise 3B - create Enrichment Map and Enhanced graphics nodes from EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-4---load-network-from-ndex"><i class="fa fa-check"></i>Exercise 4 - Load network from NDex</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1d-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1d (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#data-4"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#bonus---automation.-3"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#running-example-notebooks-in-local-rstudio"><i class="fa fa-check"></i>Running example notebooks in local RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 4 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-in-depth-analysis-of-networks-and-pathways.html"><a href="module-4-in-depth-analysis-of-networks-and-pathways.html"><i class="fa fa-check"></i>Module 4: In-depth Analysis of Networks and Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html"><i class="fa fa-check"></i>Module 4 Lab: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#data-download-the-following-files-on-your-computer-before-starting-the-practical-lab."><i class="fa fa-check"></i>Data: download the following files on your computer before starting the practical lab.</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-1-use-the-reactome-functional-interaction-fi-network"><i class="fa fa-check"></i>Exercise 1: Use the Reactome Functional Interaction (FI) Network</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-describe-the-size-and-composition-of-the-network"><i class="fa fa-check"></i>Question 1: Describe the size and composition of the network?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-2-after-clustering-how-many-modules-are-there"><i class="fa fa-check"></i>Question 2: After clustering, how many modules are there?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#query-information-about-the-interaction-between-2-genes"><i class="fa fa-check"></i>Query information about the interaction between 2 genes:</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-3-what-are-the-most-significant-pathways-in-each-module"><i class="fa fa-check"></i>Question 3: What are the most significant pathways in each module?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#set-the-size-of-the-nodes-proportional-to-the-mutation-frequencies-in-each-cancer"><i class="fa fa-check"></i>Set the size of the nodes proportional to the mutation frequencies in each cancer</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#play-around-with-the-styles-change-transparency-and-colors"><i class="fa fa-check"></i>Play around with the styles: change transparency and colors</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-pie-chart"><i class="fa fa-check"></i>Create a pie chart</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-subnetwork"><i class="fa fa-check"></i>Create a subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#fetch-cancer-drugs-on-the-created-subnetwork"><i class="fa fa-check"></i>Fetch Cancer drugs on the created subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#save-the-network-as-an-image-for-publication"><i class="fa fa-check"></i>Save the network as an image for publication</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2a-explore-reactome-pathways"><i class="fa fa-check"></i>Exercise 2a: Explore Reactome Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2b-pathway-enrichment-analysis-using-a-simple-gene-list"><i class="fa fa-check"></i>Exercise 2b: Pathway enrichment analysis using a simple gene list</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-what-are-the-most-significant-biological-pathways-based-on-the-fdr"><i class="fa fa-check"></i>Question 1: What are the most significant biological pathways based on the FDR?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#answer-to-question-1"><i class="fa fa-check"></i>Answer to Question 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2c-pathway-based-analysis-using-a-rank-gene-list-gsea"><i class="fa fa-check"></i>Exercise 2c: Pathway-based analysis using a rank gene list (GSEA)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#automation-for-advanced-users"><i class="fa fa-check"></i>Automation ( for advanced users)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#reference-guide-bonus-exercises"><i class="fa fa-check"></i>Reference guide /bonus exercises:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-gene-function-prediction.html"><a href="module-5-gene-function-prediction.html"><i class="fa fa-check"></i>Module 5: Gene Function Prediction</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html"><i class="fa fa-check"></i>Module 5 Lab: GeneMANIA (Cytoscape version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-1-searching-genemania-with-single-gene"><i class="fa fa-check"></i>EXERCISE 1: Searching GeneMANIA with single gene</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#answers"><i class="fa fa-check"></i>ANSWERS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-2-searching-genemania-with-gene-list"><i class="fa fa-check"></i>EXERCISE 2: Searching GeneMANIA with gene list</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-3-searching-genemania-with-mixed-gene-list"><i class="fa fa-check"></i>EXERCISE 3: Searching GeneMANIA with mixed gene list</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#genemania-definitions"><i class="fa fa-check"></i>GeneMANIA DEFINITIONS:</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-4-optional-discover-the-stringapp"><i class="fa fa-check"></i>EXERCISE 4 (OPTIONAL): Discover the stringApp</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#more-string-information-and-tutorials"><i class="fa fa-check"></i>More STRING information and tutorials:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html"><i class="fa fa-check"></i>Module 5 Lab: GeneMANIA (web version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-4-step5"><i class="fa fa-check"></i>Exercise 3 - STEP 4/ STEP5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-cell-cell-communication.html"><a href="module-6-cell-cell-communication.html"><i class="fa fa-check"></i>Module 6: Cell Cell Communication</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-cell-cell-communication.html"><a href="module-6-cell-cell-communication.html#module-6-lecture-cell-cell-communication."><i class="fa fa-check"></i>Module 6 lecture : Cell-Cell Communication.</a></li>
<li class="chapter" data-level="" data-path="module-6-cell-cell-communication.html"><a href="module-6-cell-cell-communication.html#scrna-lab-praticals"><i class="fa fa-check"></i>scRNA lab praticals</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html"><i class="fa fa-check"></i>Module 6 lab 1: scRNA PBMC</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#introduction-2"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#pmbc3k-seurat-pipeline"><i class="fa fa-check"></i>Pmbc3k Seurat Pipeline</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#load-libraries"><i class="fa fa-check"></i>load libraries</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#load-the-pbmc-dataset"><i class="fa fa-check"></i>Load the PBMC dataset</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#process-the-dataset"><i class="fa fa-check"></i>Process the dataset</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#assign-cell-type-identity-to-clusters"><i class="fa fa-check"></i>Assign cell type identity to clusters</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#find-differentially-expressed-features-cluster-biomarkers"><i class="fa fa-check"></i>Find differentially expressed features (cluster biomarkers)</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#create-gene-list-for-each-cluster-to-use-with-gprofiler"><i class="fa fa-check"></i>Create Gene list for each cluster to use with g:Profiler</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#tutorial_start"><i class="fa fa-check"></i>Data (gene lists for each cluster)</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#run-pathway-enrichment-analysis-using-gprofiler"><i class="fa fa-check"></i>Run pathway enrichment analysis using g:Profiler</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#create-an-enrichment-map-in-cytoscape"><i class="fa fa-check"></i>Create an enrichment map in Cytoscape</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#gsea-from-pseudobulk"><i class="fa fa-check"></i>GSEA from pseudobulk</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#pseudobulk-creation-differential-expression-and-rank-file"><i class="fa fa-check"></i>pseudobulk creation, differential expression and rank file</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#run-gsea"><i class="fa fa-check"></i>run GSEA:</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#create-an-enrichmentmap"><i class="fa fa-check"></i>Create an EnrichmentMap:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html"><i class="fa fa-check"></i>Module 6 lab 2- scRNA Glioblastoma</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#introduction-3"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#data-5"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#overview"><i class="fa fa-check"></i>Overview</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#can-module8-exercise-1"><i class="fa fa-check"></i>Part 1 - run g:Profiler [OPTIONAL]</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-1---launch-gprofiler.-1"><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-2---input-query-1"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-3---adjust-parameters.-1"><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-4---run-query-1"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-5---explore-the-results.-1"><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-6-expand-the-stats-tab-1"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-7-save-the-results-1"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-8-optional-but-recommended-1"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#exercise-2"><i class="fa fa-check"></i>Part 2 - Cytoscape/EnrichmentMap [OPTIONAL]</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#goal-of-the-exercise-6"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#data-6"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#enrichmentmap-2"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#description-of-this-exercise-1"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#start-the-exercise-3"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-1-2"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-2-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-3-explore-detailed-results"><i class="fa fa-check"></i>Step 3: Explore Detailed results</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-4-optional-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Step 4 [OPTIONAL]: AutoAnnotate the enrichment map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#exercise-3"><i class="fa fa-check"></i>Part 3 - Master map using multiple datasets</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#data-7"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#start-the-exercise-4"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-1-3"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-2-3"><i class="fa fa-check"></i>Step 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html"><i class="fa fa-check"></i>Module 6 lab 3: cellPhoneDB</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#cell-cell-communication-in-scrna-cellphonedb"><i class="fa fa-check"></i>Cell-Cell communication in scRNA: CellPhoneDB</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#presentation"><i class="fa fa-check"></i>Presentation</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#method"><i class="fa fa-check"></i>Method</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#examining-the-results"><i class="fa fa-check"></i>Examining the results</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#visualization-using-cytoscape"><i class="fa fa-check"></i>Visualization using Cytoscape</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#dataset-and-references"><i class="fa fa-check"></i>Dataset and references</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#dataset_prep"><i class="fa fa-check"></i>Dataset preprocessing and running CellPhoneDB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html"><i class="fa fa-check"></i>Module 6 lab 4: NEST</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#cell-cell-communication-ccc-in-spatial-transcriptomics-using-nest"><i class="fa fa-check"></i>Cell-Cell Communication (CCC) in spatial transcriptomics using NEST</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#nest-neural-network-on-spatial-transcriptomics"><i class="fa fa-check"></i>NEST (NEural network on Spatial Transcriptomics)</a></li>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#how-to-run-nest"><i class="fa fa-check"></i>How to run NEST</a></li>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#practical-lab-pancreatic-ductal-adenocarcinoma-pdac"><i class="fa fa-check"></i>Practical lab : Pancreatic Ductal Adenocarcinoma (PDAC)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-7-review-of-the-tools.html"><a href="module-7-review-of-the-tools.html"><i class="fa fa-check"></i>Module 7: Review of the tools</a>
<ul>
<li class="chapter" data-level="" data-path="module-7-review-of-the-tools.html"><a href="module-7-review-of-the-tools.html#final-slides"><i class="fa fa-check"></i>Final slides</a></li>
<li class="chapter" data-level="" data-path="module-7-review-of-the-tools.html"><a href="module-7-review-of-the-tools.html#integrated-assignment"><i class="fa fa-check"></i>Integrated assignment</a></li>
<li class="chapter" data-level="" data-path="module-7-review-of-the-tools.html"><a href="module-7-review-of-the-tools.html#integrated-assignment-bonus"><i class="fa fa-check"></i>Integrated assignment bonus</a></li>
<li class="chapter" data-level="" data-path="module-7-review-of-the-tools.html"><a href="module-7-review-of-the-tools.html#clusterprofiler"><i class="fa fa-check"></i>ClusterProfiler</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html"><i class="fa fa-check"></i>Module 7 Integrated Assignment</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-genemania"><i class="fa fa-check"></i>Answers GeneMANIA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html"><i class="fa fa-check"></i>Module 7 Integrated Assignment Bonus - Automation</a>
<ul>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html#goal-of-the-exercise-7"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html"><i class="fa fa-check"></i>Optional Module 8: Regulatory Network Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html"><i class="fa fa-check"></i>Optional Module 8 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#goal-of-this-practical-lab-3"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation"><i class="fa fa-check"></i>Exercise 3 (optional): Practice building enrichment maps and auto-annotation</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO genes on the distal enrichment map.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4a-post-analysis"><i class="fa fa-check"></i>step 4a: post analysis:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip"><i class="fa fa-check"></i>Exercise 5: Learning how to run MEME-chip from the MEME suite (https://meme-suite.org/meme/tools/meme-chip)</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#format-the-data"><i class="fa fa-check"></i>Format the Data</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5a-download-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Download sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4."><i class="fa fa-check"></i>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#end-of-lab"><i class="fa fa-check"></i>End of Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html"><i class="fa fa-check"></i>Optional Module 8 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#iregulon-lab"><i class="fa fa-check"></i>iRegulon lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-3"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-1.-detect-regulons-from-co-expressed-genes"><i class="fa fa-check"></i>Exercise 1. Detect regulons from co-expressed genes</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#skills-learned-in-this-exercise"><i class="fa fa-check"></i>Skills learned in this exercise:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape."><i class="fa fa-check"></i>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#use-our-precomputed-iregulon-results"><i class="fa fa-check"></i>Use our precomputed iRegulon results:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-iregulon"><i class="fa fa-check"></i>Notes about iRegulon:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-cytoscape"><i class="fa fa-check"></i>Notes about Cytoscape:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-3.-use-enrichr-with-the-prostate-gene-list."><i class="fa fa-check"></i>Exercise 3. Use Enrichr with the prostate gene list.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-4"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-1"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#end-of-practical-lab"><i class="fa fa-check"></i>end of practical lab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html"><i class="fa fa-check"></i>Optional Module 8 Lab 3: Automated Enrichment and Visualisation Lab using <code>clusterProfiler</code></a>
<ul>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#clusterprofiler-lab"><i class="fa fa-check"></i><code>clusterProfiler</code> lab</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#goal-5"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#supported-analysis"><i class="fa fa-check"></i>Supported Analysis</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#install-and-load-packages"><i class="fa fa-check"></i>Install and load packages</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#exercise-1a.-over-representation-analysis"><i class="fa fa-check"></i>Exercise 1a. Over representation analysis</a>
<ul>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#data-for-enrichment-using-clusterprofiler"><i class="fa fa-check"></i>Data for enrichment using <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#data-for-over-representation-analysis-using-clusterprofiler"><i class="fa fa-check"></i>Data for over representation analysis using <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#perform-go-over-representation-analysis"><i class="fa fa-check"></i>Perform GO over representation analysis</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#results-of-go-over-representation-analysis"><i class="fa fa-check"></i>Results of GO over representation analysis</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#input-options-for-enrichgo"><i class="fa fa-check"></i>Input options for <code>enrichGO()</code>:</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#simplify-enrichgo-results"><i class="fa fa-check"></i>Simplify <code>enrichGO()</code> results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#exercise-1b.-visualise-the-results-of-go-over-representation-analysis"><i class="fa fa-check"></i>Exercise 1b. Visualise the results of GO over representation analysis</a>
<ul>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#barplot"><i class="fa fa-check"></i>Barplot</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#dotplot"><i class="fa fa-check"></i>Dotplot</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#enrichment-map"><i class="fa fa-check"></i>Enrichment Map</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#upset-plot"><i class="fa fa-check"></i>Upset plot</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#details-about-the-input-arguments-for-enrichgo"><i class="fa fa-check"></i>Details about the input arguments for <code>enrichGO()</code></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#a-note-on-supported-organisms"><i class="fa fa-check"></i>A note on supported organisms</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#exercise-2a-gene-set-enrichment-analysis"><i class="fa fa-check"></i>Exercise 2a: Gene set enrichment analysis</a>
<ul>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#data-for-running-gene-set-enrichment-analysis-in-clusterprofiler"><i class="fa fa-check"></i>Data for running gene set enrichment analysis in <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#perform-go-gene-set-enrichment-analysis"><i class="fa fa-check"></i>Perform GO gene set enrichment analysis</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#results-of-go-gene-set-enrichment-analysis"><i class="fa fa-check"></i>Results of GO gene set enrichment analysis</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#input-options-for-gsego"><i class="fa fa-check"></i>Input options for <code>gseGO()</code></a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#details-about-the-input-arguments-for-gsego"><i class="fa fa-check"></i>Details about the input arguments for <code>gseGO()</code></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#exercise-2b.-visualise-the-results-of-gene-set-enrichment-analysis"><i class="fa fa-check"></i>Exercise 2b. Visualise the results of gene set enrichment analysis</a>
<ul>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#dotplot-1"><i class="fa fa-check"></i>Dotplot</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#ridgeline-plot"><i class="fa fa-check"></i>Ridgeline plot</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i>Running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#enrichment-map-1"><i class="fa fa-check"></i>Enrichment Map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#what-next"><i class="fa fa-check"></i>What next?</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#explore-other-features-of-clusterprofiler"><i class="fa fa-check"></i>Explore other features of <code>clusterProfiler</code></a>
<ul>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#bonus---try-it-yourself"><i class="fa fa-check"></i>Bonus - Try it yourself:</a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#ontologies-and-pathway-databases-supported-by-clusterprofiler"><i class="fa fa-check"></i>Ontologies and pathway databases supported by <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="" data-path="clusterprofiler_optionallab.html"><a href="clusterprofiler_optionallab.html#all-publications-describing-clusterprofiler-can-be-found-here"><i class="fa fa-check"></i>All publications describing <code>clusterProfiler</code> can be found here:</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a  href=""><img src="./images/sponsors_2023.png" width="200"></a><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway and Network Analysis of -Omics Data ( November 2024 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genemania_cytoscape" class="section level1 hasAnchor">
<h1>Module 5 Lab: GeneMANIA (Cytoscape version)<a href="genemania_cytoscape.html#genemania_cytoscape" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<p><em><font color="#827e9c">By Quaid Morris and Veronique Voisin </font></em></p>
<div id="goal-of-this-practical-lab-1" class="section level2 hasAnchor">
<h2>Goal of this practical lab<a href="genemania_cytoscape.html#goal-of-this-practical-lab-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create GeneMANIA networks starting from a single gene to predict its function or starting from a gene list. Explore and understand the main output features of GeneMANIA such as the network composition or the enriched functions. This practical consists of 3 exercises.</p>
<p>Before starting the exercises,download the files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work
directory.
</p>
</div>
<ul>
<li><p><a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a></p></li>
<li><p><a href="./Module6/genemania/data/mixed_gene_list.txt">Mixed_gene_list.txt</a></p></li>
<li><p><a href="./Module6/genemania/data/CYB11B_pearson_correlation_prostate.txt">CYP11B_pearson_correlation_prostate.txt</a></p></li>
</ul>
<div class="rmd-note">
<p>
Network layouts are flexible and can be rearranged. What you see when
you perform these exercises may not be identical to what you see in the
tutorial, or what you have seen other times that you have performed the
exercises. Exact layouts and predictions can also be affected by updates
to the networks database that GeneMANIA uses. However it is expected
that the network weights and predicted genes will be similar to those
shown here.
</p>
</div>
</div>
<div id="exercise-1-searching-genemania-with-single-gene" class="section level2 hasAnchor">
<h2>EXERCISE 1: Searching GeneMANIA with single gene<a href="genemania_cytoscape.html#exercise-1-searching-genemania-with-single-gene" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Imagine that you are interested in exploring the function of the human GRN gene: GRN returned as the strongest hit from your omics experiment but not much information about this gene is available in functional databases. Use GeneMANIA to identify its predicted function as well as potential interaction partners.</p>
<p><strong>Skills</strong>:</p>
<ul>
<li>GeneMANIA Single Gene search</li>
<li>Navigating Search Results</li>
<li>Exploring available Genes features</li>
<li>Rerun a new analysis using a single gene or multiple genes queried from the network.</li>
</ul>
<p><strong>Steps</strong><br></p>
<ol style="list-style-type: decimal">
<li><p>Open Cytoscape.</p></li>
<li><p>In the network tab Locate the Network search bar located at the top of the <em>Control Panel</em>. Make sure that the database selected is GeneMANIA<br><img src="Module6/genemania/images/Genemania.png" width="75%" /></p></li>
<li><p>In the search window, ensure that the model organism is set to <em>Homo sapiens</em> <img src="Module6/genemania/images/Up.png" alt="homo" />.</p></li>
<li><p>Enter the following gene in the GeneMANIA search bar: GRN</p></li>
<li><p>Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results. <br><img src="Module6/genemania/images/gc_1.1.png" alt="gc_1.1.png" width="750" /></p></li>
<li><p>When your search results load, examine the network. Genes part of the query set are indicated in black, related genes added by GeneMANIA are represented in gray, and colored links represent the interactions that connect the nodes (genes). <br><img src="Module6/genemania/images/GC2.png" alt="GC2.png" width="750" /></p></li>
</ol>
<div class="rmd-tip">
<p>
zoom in and zoom out using trackpad or mouse scrolling up and
down.
</p>
</div>
<ol start=7 type="1">
<li>
Locate the <em>Functions</em> summary tab in Results Panel. <br><img src="Module6/genemania/images/GC3.png" alt="GC3.png" width="750" />
</ol>
<p><strong>Questions</strong>:<br>
* What are the functions significantly associated with this network? <br>
* GRN is the central node of this network: which function would you predict for GRN? <br>
* How well did GeneMANIA perform? (hints: use GeneCards (<a href="http://www.genecards.org/" class="uri">http://www.genecards.org/</a>) , PubMed (<a href="http://www.ncbi.nlm.nih.gov/pubmed/" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/</a>))?</p>
<div id="answers" class="section level3 hasAnchor">
<h3>ANSWERS<a href="genemania_cytoscape.html#answers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Question</strong> What are the functions significantly associated with this network? <br>
<strong>Answer</strong> the list of the functions associated with the network are listed in the above screenshot. The top 2 pathways are “vacuolar lumen” and “primary lysosome” and are significant under a FDR threshold less than 0.005.</p>
<p><strong>Question</strong> GRN is the central node of this network: which function would you predict for GRN?
<strong>Answer</strong> : a function related to lysosome and vacuole</p>
<p><strong>Question</strong> How well did GeneMANIA perform (hints: use GeneCards (<a href="http://www.genecards.org/" class="uri">http://www.genecards.org/</a>) , PubMed (<a href="http://www.ncbi.nlm.nih.gov/pubmed/" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/</a>))?<br>
<strong>Answer</strong>
The top functions predicted by GeneMANIA for GRN were related to lysosome and vacuole. A pubmed search could confirm these results: “We experimentally verified that granulin precursor (GRN) gene, whose mutations cause frontotemporal lobar degeneration, is involved in lysosome function.” (Transcriptional gene network inference from a massive dataset elucidates transcriptome organization and gene function. Belcastro et al. Nucleic Acids Res. 2011 Nov 1;39(20):8677-88. 2011. <a href="PMID:21785136" class="uri">PMID:21785136</a>)</p>
<ol start=8 type="1">
<li>
Locate the genes with the strongest associations with GRN.
</ol>
<div class="rmd-tip">
<p>
These genes are the largest nodes in the network.
</p>
</div>
<p><strong>Answer is SLPI and SORT1</strong></p>
<ol start=9 type="1">
<li>
Re-run an analysis by adding SORT1, SLPI to the search. Type ‘SORT1’ and ‘SLPI’ in the search box that already contains ‘GRN’ (one gene per line). Click on the search button.
</ol>
<p><br><img src="Module6/genemania/images/gc_1.9.png" alt="gc_1.9.png" width="750" /></p>
<p><strong>Question</strong>:Which functions are associated with this new network?</p>
<p><img src="Module6/genemania/images/GC9b.png" alt="GC9b.png" width="750" /></p>
<p><img src="Module6/genemania/images/GC9c.png" alt="GC9c.png" width="750" /></p>
<p><strong>Biological interpretation of the results:</strong></p>
<p><strong>A paper describing the interaction between GRN and SORT1 and demonstrates how finding related genes could be relevant for elaborating therapy:</strong></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3929086/">Targeted manipulation of the sortilin–progranulin axis rescues progranulin haploinsufficiency. Lee et al. Hum Mol Genet. 2014 March 15; 23(6): 1467–1478. PMCID:PMC3929086</a><br>
“Progranulin (GRN) mutations causing haploinsufficiency are a major cause of frontotemporal lobar degeneration (FTLD-TDP). Recent discoveries demonstrating sortilin (SORT1) is a neuronal receptor for PGRN endocytosis and a determinant of plasma PGRN levels portend the development of enhancers targeting the SORT1–PGRN axis. We demonstrate the preclinical efficacy of several approaches through which impairing PGRN’s interaction with SORT1 restores extracellular PGRN levels. “</p>
<p><img src="Module6/genemania/images/GM11.png" /></p>
<ol start=10 type="1">
<li>
<p>Save the network as an image by clicking on <strong>File</strong>, <strong>Export</strong>, <strong>Network to Image…</strong> and setting the <strong>Export File Format</strong> to “PDF(*.pdf)”.<br><img src="Module6/genemania/images/GC10.png" alt="GC10.png" width="500" /></p>
<hr />
<hr />
</div>
</div>
<div id="exercise-2-searching-genemania-with-gene-list" class="section level2 hasAnchor">
<h2>EXERCISE 2: Searching GeneMANIA with gene list<a href="genemania_cytoscape.html#exercise-2-searching-genemania-with-gene-list" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To start this exercise, you need to download the <a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a> file and save it on your computer.</p>
<p>For this exercise, you are working with a list of 30 prostate cancer genes. This list was downloaded from the cBioPortal website (<a href="http://www.cbioportal.org/" class="uri">http://www.cbioportal.org/</a>). The cBioPortal for Cancer Genomics stores genomic data from large scale, integrated cancer genomic data sets. During this exercise, you will explore the types of networks that have been used to create the GeneMANIA network from the prostate cancer gene list and you will see how changing input parameters can affect the results. The last step of the exercise consists of uploading a custom network which is a list of genes that are positively correlated with CYP11B1 in mRNA expression data of 94 prostate cancer samples (<a href="http://www.cbioportal.org/" class="uri">http://www.cbioportal.org/</a>) .</p>
<p><strong>Skills</strong>: <br></p>
<ul>
<li>GeneMANIA search using a gene list;</li>
<li>Navigating Search Results;</li>
<li>Exploring Networks and advanced options;</li>
<li>Uploading a custom network.</li>
</ul>
<p><strong>Steps</strong><br></p>
<ol style="list-style-type: decimal">
<li><p>Open Cytoscape.</p></li>
<li><p>Locate the GeneMANIA search window located on the left side in <em>Control Panel</em>.</p></li>
<li><p>Copy and paste genes in the file <a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a></p>
<ul>
<li>Make sure that the parameter ‘Max resultant genes’ is set to ‘20’ by clicking on the menu button <img src="Module6/genemania/images/options.png" alt="options" /> at the right side of the search box and selecting ‘Customise advanced options’.</li>
</ul></li>
<li><p>Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results. <br><img src="Module6/genemania/images/gc_2_4.png" alt="gc_2_4.png" width="750" />
``</p></li>
<li><p>When your search results load, examine the network. Query genes are indicated in black, related genes added by GeneMANIA are represented in gray, and colored links represent the interactions that connect the nodes (genes). Move nodes around by selecting them with a mouse to investigate how they are connected.<br><img src="Module6/genemania/images/GC2_5.png" alt="GC2_5.png" width="750" /></p></li>
<li><p>Click any link (edge) connecting two nodes to highlight information about it. The information about the interaction is display in the <em>Edge Table</em> located in <em>Table Panel</em> (at the bottom) in the <em>networks</em> and <em>data type</em> columns.</p>
<ul>
<li><strong>Note</strong>: Clicking on an edge between 2 nodes will display information about all interaction networks that connect these 2 nodes.</li>
<li>It indicates the reference (publication) for these interactions.</li>
<li>The colors indicate the type of interaction (co-expression, shared protein domains, co-localization, physical interactions and predicted). <br><img src="Module6/genemania/images/gc_2.6.png" alt="gc_2.6.png" width="750" /></li>
</ul></li>
<li><p>Locate and expand the ‘Networks’ summary tab in <em>Results Panel</em> (on the right) and look at what data has been used to create the network and predictions.</p>
<ul>
<li><strong>Note</strong> that Co-expression (purple colored lines, weight over 25%) and Shared protein domains (lightgold colored lines, weight over 30%) influence the results the most, but Co-localization (blue colored lines), Physical interactions (salmon colored lines) and Predicted (orange) data are also included.</li>
<li>At the top of the Networks summary tab, use the menu button <img src="Module6/genemania/images/options.png" alt="options" /> and try Expand “All, then “Top-Level” and “None” to get information about the sources of the different networks. <br><img src="Module6/genemania/images/GC2_7.png" alt="GC2_7.png" width="550" /></li>
</ul></li>
</ol>
<div class="rmd-tip">
<p>
The observations of the number of connections makes it easier to
understand why co-expression and shared protein domains have the highest
percent weight for this network: they are helping to connect more genes
than physical interactions and predicted interactions; A higher weight
means that this network contributed more to finding related genes.
</p>
</div>
<ol start=8 type="1">
<li>
Highlight all connections corresponding to each network by clicking the name of each network category.
</ol>
<ul>
<li>Click on “Shared protein domains” and see which genes are connected by predicted protein protein interaction. <br><img src="Module6/genemania/images/GC2_8a.png" alt="GC2_8a.png" width="750" /></li>
<li>You can do the same for “Co-localization” , “Co-expression” and “Physical interactions”. <br><img src="Module6/genemania/images/GC2_8b.png" alt="GC2_8b.png" width="750" /></li>
</ul>
<ol start=9 type="1">
<li>
Locate the Functions summary tab and look at what functions were significantly enriched in this list of prostate genes.
</ol>
<ul>
<li>The top pathway with the strongest enrichments is: “oxidoreductase activity, acting on CH-OH group of donors” with 28 genes in the prostate cancer list overlapping with this pathway.</li>
<li>The FDR is equal to 6.4e-46.<br><img src="Module6/genemania/images/GC2_9.png" alt="GC2_9.png" width="750" /></li>
</ul>
<p><strong>Question</strong>:<br>“Shared protein domains” is an important part of the network. What would the GeneMANIA results be if we didn’t include this source when we ran GeneMANIA search?</p>
<ul>
<li>Go back to the ‘Network’ tab on the right side of the Cytoscape window to find the GeneMania search bar.</li>
<li>Click on the option menu button <img src="Module6/genemania/images/options.png" alt="options" /> which is located at the right of the search box.</li>
<li>Uncheck ‘Shared protein domains’ and click on a point outside the box to close it.</li>
<li>Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" />.</li>
<li>Explore the results.<br><img src="Module6/genemania/images/GC2_10a.png" alt="GC2_10a.png" width="750" /></li>
</ul>
<p><strong>Answer</strong> <br>If “shared protein domain” is removed, the relationships between the nodes are primarily from the Co-expression, Co-localization, Predicted and Physical interactions networks. The genes added to the network are different compared to the first network created with “Shared protein domain”. <br><img src="Module6/genemania/images/GC2_10b.png" alt="GC2_10b.png" width="750" /></p>
<p><strong>Question</strong>:<br>Locate the Functions summary tab in <em>Results Panel</em> and look at what functions were significantly enriched with these new settings.</p>
<p><strong>Answer</strong> <br>With the new settings, “steroid biosynthetic process” is the new top enriched pathway. <br><img src="Module6/genemania/images/GC2_11.png" alt="GC2_11.png" width="500" /></p>
<ol start=10 type="1">
<li>
Try to modify additional parameters like <em>Max Resultant Genes</em> or <em>Network Weighting</em> and look at how the changes you made influenced the results.
</ol>
<hr />
<hr />
</div>
<div id="exercise-3-searching-genemania-with-mixed-gene-list" class="section level2 hasAnchor">
<h2>EXERCISE 3: Searching GeneMANIA with mixed gene list<a href="genemania_cytoscape.html#exercise-3-searching-genemania-with-mixed-gene-list" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To start this exercise, you need to download the <a href="./Module6/genemania/data/mixed_gene_list.txt">Mixed_gene_list.txt</a> file and save it on your computer.</p>
<p>For this exercise, you are working on a gene list created by combining 3 user defined gene lists available from the cBioportal (<a href="http://www.cbioportal.org" class="uri">http://www.cbioportal.org</a>). It contains genes implicated in the DNA damage response, the PI3K-AKT-mTOR signaling pathway and Folate transport. This list is representative of a gene list obtained from transcriptomics data. During this exercise, we will first characterize our gene list based on functions and then we will add potential drug and microRNAs targeting genes in the network, and we will save the report.</p>
<p><strong>Skills</strong>:</p>
<ul>
<li>GeneMANIA search using a gene list;</li>
<li>Navigating Search Results;</li>
<li>Exploring Functions;</li>
<li>Adding attributes;</li>
<li>Create a report.</li>
</ul>
<p><strong>Steps</strong><br></p>
<ol style="list-style-type: decimal">
<li><p>Before performing the next GeneMANIA search make sure the GeneMANIA parameters are set back to the default values. <br><img src="Module6/genemania/images/default_params.png"  width="500" /></p></li>
<li><p>Open Cytoscape and locate the GeneMANIA search window located on the left side in <em>Control Panel</em>.</p></li>
<li><p>In the search window, ensure that the model organism is set to <em>Homo sapiens</em> <img src="Module6/genemania/images/Up.png" alt="homo" /> .</p></li>
<li><p>Copy and paste genes in the file <a href="./Module6/genemania/data/mixed_gene_list.txt">Mixed_gene_list.txt</a>. Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results. Explore the network.<br><img src="Module6/genemania/images/gc_3_2.png" alt="gc_3_2.png" width="500" /></p></li>
<li><p>Locate the Functions summary tab in <em>Result Panel</em> and look at functions returned by GeneMANIA. <br><img src="Module6/genemania/images/GC3_4.png" alt="GC3_4.png" width="500" /></p></li>
<li><p>In the functions summary tab, check some functions to color genes included in these functions. To follow this tutorial, you can for example color the “DNA recombination” , “response to insulin” functions.<br><img src="Module6/genemania/images/GC3_4a.png" alt="GC3_4a.png" width="500" /><br><img src="Module6/genemania/images/GC3_4b.png" alt="GC3_4b.png" width="500" /></p></li>
<li><p>Color genes according to their GeneMANIA defined functions:</p>
<ul>
<li>Go to the <strong>Control Panel</strong> tabs located on the right side of the Cytoscape window and select the <strong>Style</strong> tab.<br />
</li>
<li>In the <strong>Node</strong> panel, expand the <strong>Fill Color</strong> tab.</li>
<li>Set <strong>Column</strong> to <strong>annotation name</strong>.<br><img src="Module6/genemania/images/gc3_5a.png" alt="gc3_5a.png" width="500" /></li>
<li>Locate “DNA recombination”.</li>
<li>Double click on the white space at the right side of the box and click on the 3 dots <img src="Module6/genemania/images/options2.png" alt="options2" />. A <strong>Colors</strong> box appears.</li>
<li>Choose a color of your choice and click on <strong>OK</strong>.<br><img src="Module6/genemania/images/GC3_5.png" alt="GC3_5.png" width="500" /></li>
<li>Locate “response to insulin”. Double click on the white space at the right side of the box and click on the 3 buttons menu. A <strong>Colors</strong> box appears.</li>
<li>Choose a color of your choice and click on <strong>OK</strong>.<br><img src="Module6/genemania/images/GC3_5b.png" alt="GC3_5b.png" width="750" /></li>
</ul></li>
<li><p>Locate our favorite gene PDPK1 on the network.</p>
<ul>
<li>Click on the icon <em>First Neighbor of Selected Nodes</em> <img src="Module6/genemania/images/neighbour.png" alt="neighbour" />. It will highlight this gene and all its connections. <br><img src="Module6/genemania/images/GC3_6.png" alt="GC3_6.png" width="750" /></li>
<li>Click on the icon <em>From Selected Nodes, all Edges</em> <img src="Module6/genemania/images/newnetwork.png" alt="new network" /> to create a subnetwork.<br><img src="Module6/genemania/images/GC3_6b.png" alt="GC3_6b.png" width="750" /></li>
<li>The resultsing subnetwork will only have the selected nodes from the first network<br><img src="Module6/genemania/images/GC3_6c.png" alt="GC3_6c.png" width="750" /></li>
</ul></li>
</ol>
<div class="rmd-tip">
<p>
copy “PDPK1” to the search box, click enter and the node will be
highlighted in yellow in the network.
</p>
</div>
<hr />
</div>
<div id="genemania-definitions" class="section level2 hasAnchor">
<h2>GeneMANIA DEFINITIONS:<a href="genemania_cytoscape.html#genemania-definitions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>What are the different networks: Definition of the types of interaction:</strong></p>
<ul>
<li><p><strong>Shared domains</strong>: Protein domain data. Two gene products are linked if they have the same protein domain. These data are collected from domain databases, such as InterPro, SMART and Pfam.</p></li>
<li><p><strong>Co-localization</strong>: Genes expressed in the same tissue, or proteins found in the same location. Two genes are linked if they are both expressed in the same tissue or if their gene products are both identified in the same cellular location.</p></li>
<li><p><strong>Co-expression</strong>: Gene expression data. Two genes are linked if their expression levels are similar across conditions in a gene expression study. Most of this data is collected from the Gene Expression Omnibus (GEO); we only collect data associated with a publication.</p></li>
<li><p><strong>Predicted</strong>: Predicted functional relationships between genes, often protein interactions. A major source of predicted data is mapping known functional relationships from another organism via orthology.</p></li>
</ul>
<p><strong>What is defined by evidence sources?:</strong></p>
<ul>
<li><strong>Evidence sources</strong> are the information contained in the multiple databases that GeneMANIA uses to establish interaction between two genes.</li>
</ul>
<p><strong>Network:</strong></p>
<ul>
<li><p><strong>Node</strong> : circle representing the genes</p></li>
<li><p><strong>Edge</strong>: line that links two nodes and represent an interaction between two genes (multiple lines correspond to multiple sources)</p></li>
<li><p><strong>Node size</strong>: Mapped to gene score, i.e. the degree to which GeneMANIA predicts the genes are related</p></li>
<li><p><strong>Thickness of edge</strong>: Strength/weight of interaction</p></li>
</ul>
<p><strong>Layout</strong> : The layout is different each time so the user can request the layout run multiple times until the user is satisfied with the result.</p>
<p><strong>in Networks tab:</strong></p>
<ul>
<li><strong>Percent weight (score)</strong> : a higher weight means that this network helped more to find related genes.</li>
</ul>
<p><strong>in Functions tab</strong> :</p>
<ul>
<li><p><strong>FDR</strong> : False discovery rate (FDR) is greater than or equal to the probability that this is a false positive.</p></li>
<li><p><strong>Coverage</strong> : (number of genes in the network with a given function) / (all genes in the genome with the function)</p></li>
</ul>
<div id="in-advanced-options" class="section level4 hasAnchor">
<h4>In advanced options:<a href="genemania_cytoscape.html#in-advanced-options" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><p><strong>Network weighting?</strong> GeneMANIA can use a few different methods to weight networks when combining all networks to form the final composite network that results from a search. The default settings are usually appropriate, but you can choose a weighting method in the advanced option panel. (more details at <a href="http://pages.genemania.org/help/" class="uri">http://pages.genemania.org/help/</a>).</p></li>
<li><p><strong>Related genes</strong> : are genes added by GeneMANIA in addition to the genes from the query. It helps to expand the network and predict function of the query gene(s).</p></li>
<li><p><strong>The attributes</strong> represent the differences sources of evidence that can be used to build the network.</p></li>
</ul>
<p><strong>Notes</strong> :</p>
<ul>
<li><p>prostate cancer gene list is “AKR1C3 AR CYB5A CYP11A1 CYP11B1 CYP11B2 CYP17A1 CYP19A1 CYP21A2 HSD17B1 HSD17B10 HSD17B11 HSD17B12 HSD17B13 HSD17B14 HSD17B2 HSD17B3 HSD17B4 HSD17B6 HSD17B7 HSD17B8 HSD3B1 HSD3B2 HSD3B7 RDH5 SHBG SRD5A1 SRD5A3 STAR”.</p></li>
<li><p>mixed gene list is AKT1 AKT1S1 AKT2 ATM ATR BRCA1 BRCA2 CHEK1 CHEK2 FANCF FOLR1 FOLR2 FOLR3 FOXO1 FOXO3 MDC1 MLH1 MLST8 MSH2 MTOR PARP1 PDPK1 PIK3CA PIK3R1 PIK3R2 PTEN RAD51 RHEB RICTOR RPTOR SLC19A1 TSC1 TSC2</p></li>
</ul>
<div class="rmd-tip">
<p>
look at GeneMANIA help pages when you run an analysis on your own
after the workshop: <a href="http://pages.genemania.org/help/"
class="uri">http://pages.genemania.org/help/</a>.
</p>
</div>
</div>
</div>
<div id="exercise-4-optional-discover-the-stringapp" class="section level2 hasAnchor">
<h2>EXERCISE 4 (OPTIONAL): Discover the stringApp<a href="genemania_cytoscape.html#exercise-4-optional-discover-the-stringapp" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://string-db.org/">stringApp</a> imports functional associations or physical interactions between protein-protein and protein-chemical pairs from STRING, Viruses.STRING, STITCH, DISEASES and from PubMed text mining into Cytoscape.
Users provide a list of one or more gene, protein, compound, disease, or PubMed queries, the species, the network type, and a confidence score and stringApp queries the database to return the matching network.</p>
<p>Currently, five different queries are supported:</p>
<ul>
<li>STRING: protein query – enter a list of protein names (e.g. gene symbols or UniProt identifiers/accession numbers) to obtain a STRING network for the proteins</li>
<li>STRING: PubMed query – enter a PubMed query and utilize text mining to get a STRING network for the top N proteins associated with the query</li>
<li>STRING: disease query – enter a disease name to retrieve a STRING network of the top N proteins associated with the specified disease</li>
<li>STITCH: protein/compound query – enter a list of protein or compound names to obtain a network for them from STITCH</li>
<li>STRING: cross-species query – choose two species to obtain a STRING network between and within the proteins of the interacting species</li>
</ul>
<p><strong>Data</strong></p>
<p>Let’s use the prostate cancer gene list that we used in exercise 1.</p>
<ul>
<li><a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a></li>
</ul>
<p><strong>Steps</strong>:<br></p>
<ol style="list-style-type: decimal">
<li>Open Cytoscape</li>
<li>Make sure stringApp is installed. Go to menu, Apps, App Store, Show App Store. Install the app if necessary.</li>
<li>In Cytoscape, locate the <strong>Network</strong> tab and select <strong>STRING</strong>, <strong>STRING: protein query</strong> by clicking the down arrow.<br></li>
</ol>
<p><img src="Module6/stringlab/SL1.png" alt="start" width="70%" /></p>
<ol style="list-style-type: decimal">
<li><p>Copy and paste the <a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a> in the blank field and click on the search button.<br></p></li>
<li><p>Observe the network that has been created. The genes from our list are connected by predicted protein-protein interactions.<br><img src="Module6/stringlab/string1.png" alt="start" width="70%" /></p></li>
<li><p>On the right side of the Cytoscape window, locate and expand the <em>STRING</em> tab.<br></p>
<ul>
<li>Make sure that the <strong>Nodes</strong> tab is selected.<br></li>
<li>Play with parameters on the top fields: <em>Glass ball effect</em>, <em>STRING style labels</em>, etc… and observe the changes on the network.<br><img src="Module6/stringlab/string2.png" alt="start" width="70%" /></li>
</ul></li>
<li><p>Optimize the layout. In Cytoscape, go to the menu bar, Layout, yFiles Organic Layout.<img src="Module6/stringlab/string3.png" alt="start" width="40%" /></p></li>
<li><p>Go back to the STRING Note tab on the right side:</p>
<ul>
<li>Select a node and look at the gene details in the <strong>Selected nodes</strong> tab.</li>
<li>Try the <strong>Functional enrichment</strong> and observe the resuls in the <strong>STRING Enrichment</strong> table located below the network.<br><img src="Module6/stringlab/string4.png" alt="start" width="70%" /></li>
</ul></li>
<li><p>Select the <strong>Edges</strong> tab.</p>
<ul>
<li>The <strong>score</strong> slide bar enables to select the interactions with the strongest prediction scores.</li>
<li>The <strong>Subscore</strong> table traces the source of the predicted interactions using several evidence scores.<br><img src="Module6/stringlab/string5.png" alt="start" width="70%" /></li>
</ul></li>
</ol>
</div>
<div id="more-string-information-and-tutorials" class="section level2 hasAnchor">
<h2>More STRING information and tutorials:<a href="genemania_cytoscape.html#more-string-information-and-tutorials" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Reference: <a href="https://apps.cytoscape.org/apps/stringapp" class="uri">https://apps.cytoscape.org/apps/stringapp</a></li>
<li>Tutorial: <a href="https://cytoscape.org/cytoscape-tutorials/protocols/stringApp/#/" class="uri">https://cytoscape.org/cytoscape-tutorials/protocols/stringApp/#/</a></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-5-gene-function-prediction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genemania_web.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/commit/master/5.1-Module5_Cytoscape_lab.Rmd",
"text": null
},
"view": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/master/5.1-Module5_Cytoscape_lab.Rmd",
"text": null
},
"download": ["CBW_Pathways.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
